.github/workflows/run-mypy.yml

```diff
@@ -0,0 +1,24 @@
+name: MyPy maltoolbox
+on: [push]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r tests/requirements.txt
+      - name: Type checking with MyPy
+        run: |
+          pip install mypy
+          mypy .
```

Comment:
Mypy has a lot of options that can be set on the cli or in a config file (pyproject.toml). I suggest that the strictest mode is enabled unless it breaks sth that we can't fix. By strict mode I mean all checks and strict-looking flags that do not allow for a lot of deviation or looseness in the code.

Reply:
Running with --strict gives 215 errors currently (no errors without --strict). I could try to solve them all, but I think it might be worth merging this PR before. What do you think?

Reply:
Makes sense. A py.typed file [should also be added](https://peps.python.org/pep-0561/#packaging-type-information) once that issue is completed.